cmd/bundle/plan.go

```diff
@@ -74,8 +73,41 @@ func newPlanCommand() *cobra.Command {
 
 		changes := phases.Diff(ctx, b)
 
+		// Count actions by type and collect formatted actions
+		createCount := 0
+		updateCount := 0
+		deleteCount := 0
+		var actions []string
+
 		for _, change := range changes {
-			cmdio.LogString(ctx, fmt.Sprintf("%s %s.%s", change.ActionType, change.Group, change.Key))
+			actionItem := fmt.Sprintf("  %s %s.%s", change.ActionType, change.Group, change.Key)
+			actions = append(actions, actionItem)
+
+			switch change.ActionType.String() {
+			case "create":
+				createCount++
+			case "update", "update_with_id":
+				updateCount++
+			case "delete":
+				deleteCount++
+			case "recreate":
+				// A recreate counts as both a delete and a create
+				deleteCount++
+				createCount++
+			}
+		}
+
+		// Print summary line and actions to stdout
+		totalChanges := createCount + updateCount + deleteCount
+		if totalChanges > 0 {
+			fmt.Printf("Plan: %d to add, %d to change, %d to delete\n", createCount, updateCount, deleteCount)
+
+			// Print all actions in the order they were processed
+			for _, action := range actions {
+				fmt.Println(action)
+			}
+		} else {
+			fmt.Println("No changes. Your infrastructure matches the configuration.")
```

Comment:
Instead of `fmt.Printf` and `fmt.Println`, we should use `fmt.Fprintf` and friends with `cmd.OutOrStdout()`. That enables redirection elsewhere and is in line with the rest of the code under `cmd`. Looking into a linter for this.